pkg/cmd/server/server.go

```diff
@@ -141,10 +145,15 @@ func (c *Config) Complete() (RunnableServer, error) {
 			return nil, fmt.Errorf("failed to create dispatcher: %w", cerr)
 		}
 
+		presharedKey := ""
+		if len(c.PresharedKey) > 0 {
```

Comment:
Why check this? Shouldn't it error before here if `len(c.PresharedKeys) == 0`?

Reply:
Not if no keys were given; remember: preshared keys are ignored if a custom auth function is specified

Reply:
Oh, gotcha.

I wonder if it makes more sense to hoist this up to where all of the other preshared key logic is.